src/wirecraft/client/__init__.py

```diff
@@ -7,154 +7,253 @@
-            self.menu()
+
+        if back_rect.collidepoint(mouse_pos) and mouse_clicked:
+            self.view = "menu"
+            self.click_handled = True
 
     def menu(self) -> None:
         self.view = "menu"
-        self.displaysurf.fill((255, 255, 255))
+        self.displaysurf.fill(WHITE)
         titlefont = pygame.font.Font(None, 150)
         buttonfont = pygame.font.Font(None, 100)
+
+        # Render title
         title = titlefont.render("Wirecraft", True, BLACK)
         title_rect = title.get_rect(center=(self.resolution.width / 2, self.resolution.height / 5))
         self.displaysurf.blit(title, title_rect)
+
+        mouse_pos = pygame.mouse.get_pos()
+        mouse_clicked = pygame.mouse.get_pressed()[0] and not self.click_handled
+
+        # Render play button
         play = buttonfont.render("Play", True, BLACK)
         play_rect = play.get_rect(center=(self.resolution.width / 2, self.resolution.height / 3))
         self.displaysurf.blit(play, play_rect)
-        if play_rect.collidepoint(pygame.mouse.get_pos()) and pygame.mouse.get_pressed()[0]:
-            self.game()
+        if play_rect.collidepoint(mouse_pos) and mouse_clicked:
+            self.view = "game"
+            self.click_handled = True
+
+        # Render settings button
         settings = buttonfont.render("Settings", True, BLACK)
         settings_rect = settings.get_rect(center=(self.resolution.width / 2, self.resolution.height / 2))
         self.displaysurf.blit(settings, settings_rect)
-        if settings_rect.collidepoint(pygame.mouse.get_pos()) and pygame.mouse.get_pressed()[0]:
-            self.settings()
+        if settings_rect.collidepoint(mouse_pos) and mouse_clicked:
+            self.view = "settings"
+            self.click_handled = True
+
+        # Render exit button
         exit = buttonfont.render("Exit", True, BLACK)
         exit_rect = exit.get_rect(center=(self.resolution.width / 2, self.resolution.height / 1.5))
         self.displaysurf.blit(exit, exit_rect)
-        if exit_rect.collidepoint(pygame.mouse.get_pos()) and pygame.mouse.get_pressed()[0]:
+        if exit_rect.collidepoint(mouse_pos) and mouse_clicked:
             pygame.quit()
             sys.exit()
 
-    def updatecam(self) -> None:
-        # TODO Add support for mouse drag and touchpad two finger drag
-        # TODO This is terrible, fix it
-        if pygame.key.get_pressed()[pygame.K_LEFT]:
-            self.camera.x -= 5
-        if pygame.key.get_pressed()[pygame.K_RIGHT]:
-            self.camera.x += 5
-        if pygame.key.get_pressed()[pygame.K_UP]:
-            self.camera.y -= 5
-        if pygame.key.get_pressed()[pygame.K_DOWN]:
-            self.camera.y += 5
+    def handle_input(self, event: pygame.event.Event):
+        if event.type == QUIT:
+            pygame.quit()
+            sys.exit()
+        elif event.type == pygame.KEYDOWN and event.key == pygame.K_ESCAPE:
+            self.view = "menu"
```

Comment:
I don't really like string "keys" for states, but this could be discussed/changed later